lib/array_equals.rb

```diff
@@ -1,5 +1,24 @@
 # Determines if the two input arrays have the same count of elements
 # and the same integer values in the same exact order
+
 def array_equals(array1, array2)
-  raise NotImplementedError
+  i = 0
+  if (array1 == nil) && (array2 == nil)
+    return true
+  elsif (array1 == nil) && (array2 != nil)
+    return false
+  elsif (array1 != nil) && (array2 == nil)
+    return false
+  elsif array1.length != array2.length
+    return false
+  else
+    array1.length.times do
```

Comment:
Ruby provides a nice shortcut for initializing and incrementing the loop index - just add |i| to the end of this statement.